xbmc/addons/AddonVersion.cpp

```diff
@@ -55,6 +55,10 @@ namespace ADDON
     }
   }
 
+  AddonVersion::AddonVersion(const char* version)
+    :AddonVersion(std::string(version ? version : ""))
+  {}
+
```

Comment:
With this I'm not really sure, think something like that could bring in hidden mistakes.
The whole Compare stuff is designed in "0.0.0" string style and usually does not expect an empty string.

Think it is about ignoring a missing Min version?

Reply:
There are 2 reasons:
1.) we often pass const char* to AddonVersion and have the check for nullptr at many places
2.) The decision that 0.0.0 is empty() should be done from the class itself, not from the caller.

I'll replace all these conditions from code in a final step in this PR, for example here: https://github.com/xbmc/xbmc/pull/16591/files#diff-e4059697a1c464a6229909f8b684c0dfR379

Reply:
Regarding to your comment: empty is replaced to 0.0.0 here:
https://github.com/xbmc/xbmc/pull/16591/commits/a30f9a6fa0f59bd5a4e311ebecfce0e76c28ad77#diff-fbb8bcc345bc27918df721fb8c97819cL30